include/seqan3/core/debug_stream/range.hpp

```diff
@@ -135,15 +155,52 @@ inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, rng
  * to avoid ambiguous function calls.
  * \endif
  */
+#if 0
 template <typename char_t, sequence sequence_t>
 inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, sequence_t && sequence)
     requires detail::debug_stream_range_guard<sequence_t>
           && (!detail::is_uint_adaptation_v<std::remove_cvref_t<std::ranges::range_reference_t<sequence_t>>>)
 {
+#else
+template <typename sequence_t>
+    requires sequence<sequence_t>
```

Comment:
How simple this can be :)

Of course, we still have the cost of ensuring the correct print order, but it can be seen in a single place at the definition of `seqan3::default_printer`.